plugins/macro/Macro/Utilities.pm

```diff
@@ -49,7 +49,7 @@ sub ai_isIdle {
 		}
 	}
 	return 1 if (AI::is('macro', 'deal'));
-	return 1 if (AI::is('NPC') && $char->args->waitingForSteps);
+	return 1 if (AI::is('NPC') && $char->args->{waitingForSteps});

```

Comment:
I agree with your opinion but this code is just to update the macro plugin to work with the latest talk npc task .
we already have eventMacro, if someone wants to update the macro plugin, feel free to this.